### PR TITLE
Add Validations And Specs To Users, Posts, Comments and Likes Models

### PR DESCRIPTION
**Pull request by sharon-odhiambo**
In this PR I created Validations and Specs for the BlogApp Models

**Below are the highlights**

Create Validations for Users name and posts counter
Create validations for users post title, likes counter and comments counter
Create unit tests for custom methods on users, posts, likes and comments
Ensure no linter errors

**Cheers and Happy Reviewing!**

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    it 'it is a positive integer' do
+      the_user.posts_counter = -20.5
+      expect(the_user).to_not be_valid
+    end
+
+    it 'it acccepts only integers' do
+      [0, 10].each do |counter|
+        the_user.posts_counter = counter
+        expect(the_user).to be_valid
+      end
```
Comment (omar-muhamad) [inline spec/models/user_spec.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.), Please add a unit test for the `most_recent_posts` method.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @post.title = 'Helo World!'
+    expect(@post).to be_valid
+  end
+  it '@post comments_counter to be an integer greater than or equal to 0' do
+    @post.comments_counter = 'one'
+    expect(@post).to_not be_valid
+  end
+  it '@post likes_counter to be an integer greater than or equal to 0' do
+    @post.likes_counter = 'one'
+    expect(@post).to_not be_valid
```
Comment (omar-muhamad) [inline spec/models/post_spec.rb:33]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.), Please add a unit test for the `most_recent_comments` method.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @sharon-odhiambo,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by sharon-odhiambo** [comment]
Hi @Omar-Muhamad , thanks for the review. I have implemented the changes.

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @sharon-odhiambo    , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x] Your project is professional 

- [x]  Your repo is professional 




 

- [x] No Linter Errors





Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sharon-odhiambo** [comment]
Hi @KayLemba , thank you for the review.

**Pull request merged**